tests/test_integration/test_delete.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from cognite.client import CogniteClient
+
+from tests.constants import IS_PYDANTIC_V2
+
+if IS_PYDANTIC_V2:
+    from omni import OmniClient
+    from omni import data_classes as dc
+else:
+    from omni_pydantic_v1 import OmniClient
+    from omni_pydantic_v1 import data_classes as dc
+
+
+def test_delete_recursive(omni_client: OmniClient, cognite_client: CogniteClient) -> None:
```

Comment:
This test demonstrates the new behavior @Juliamg

Reply:
The test looks good and demonstrates the functionality of deleting the connecting node and the edge that connects them.

What will happen in the case where Item B has an edge to another node, say Item C. Would the deletion of Item A still delete item B, or is there a check somewhere that item B has no other edges to any other nodes, which would mean that the delete operation should keep Item B, delete Item A and the edge between Item A and Item B?

Reply:
Yes, that is correct. The delete call will only delete the nodes and edges that are passed in. If there exists other edges or connected nodes, there is no-lookup to delete them.